pyresample/test/test_dask_ewa.py

```diff
@@ -0,0 +1,389 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2021
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Test EWA Dask-based resamplers."""
+
+import logging
+from unittest import mock
+import numpy as np
+import pytest
+
+try:
+    from pyproj import CRS
+except ImportError:
+    CRS = None
+
+LOG = logging.getLogger(__name__)
+
+
+def _fill_mask(data):
+    if np.issubdtype(data.dtype, np.floating):
+        return np.isnan(data)
+    elif np.issubdtype(data.dtype, np.integer):
+        return data == np.iinfo(data.dtype).max
+    else:
+        raise ValueError("Not sure how to get fill mask.")
+
+
+def get_test_data(input_shape=(100, 50), output_shape=(200, 100), output_proj=None,
```

Comment:
Can the content of this be split into multiple smaller methods?

Reply:
Yes probably. This was copied from the old tests in Satpy with very few modifications. I'll see what I can do.